library/ecp.c

```diff
@@ -767,6 +767,68 @@ int mbedtls_ecp_point_write_binary( const mbedtls_ecp_group *grp,
     return( ret );
 }
 
+#if defined(MBEDTLS_ECP_SHORT_WEIERSTRASS_ENABLED)
+static int mbedtls_ecp_sw_derive_y( const mbedtls_ecp_group *grp,
+                                    const mbedtls_mpi *X,
+                                    mbedtls_mpi *Y,
+                                    int parity_bit )
+{
+    /* y^2 = x^3 + ax + b
+     * sqrt(w) = w^((p+1)/4) mod p   (for prime p where p = 3 mod 4)
```

Comment:
I think it would be helpful to add here information that `w = y^2`.
I'm not familiar with this topic and I was asking myself what is `w` and looking on the internet. It is described below together with computations.

Reply:
Good point. Could be as simple as:
```
    /* w = y^2 = x^3 + ax + b
     * y = sqrt(w) = w^((p+1)/4) mod p   (for prime p where p = 3 mod 4)
```

Reply:
:rocket: